sources/packages/backend/libs/services/src/notifications/notification/notification-actions.service.ts

```diff
@@ -49,29 +55,41 @@ export class NotificationActionsService {
     auditUserId: number,
     entityManager?: EntityManager,
   ): Promise<void> {
-    const templateId = await this.notificationMessageService.getTemplateId(
-      NotificationMessageType.StudentFileUpload,
-    );
-    const notificationToSend = {
-      userId: notification.userId,
-      messageType: NotificationMessageType.StudentFileUpload,
-      messagePayload: {
-        email_address: this.gcNotifyService.ministryToAddress(),
-        template_id: templateId,
-        personalisation: {
-          givenNames: notification.firstName ?? "",
-          lastName: notification.lastName,
-          dob: getDateOnlyFormat(notification.birthDate),
-          applicationNumber: notification.applicationNumber,
-          documentPurpose: notification.documentPurpose,
-          date: this.getDateTimeOnPSTTimeZone(),
+    const notificationDetails =
+      await this.notificationMessageService.getNotificationMessageDetails(
+        NotificationMessageType.StudentFileUpload,
+      );
+    if (!notificationDetails.emailContacts?.length) {
+      this.logger.error(
+        `Email template id ${NotificationMessageType.StudentFileUpload} requires a configured email to be sent.`,
+      );
+      return;
+    }
+    const ministryNotificationsToSend = [];
```

Comment:
The use of a `map` would make the method easier to read, IMO. Is there a reason to chose `for` for this scanrio?

Reply:
I remember you explaining me this in call earlier. At the time, I didn't get it. Now I got it 😅 . Will update it. Thanks